language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddCheckCodeAction.java

```diff
@@ -67,8 +68,14 @@ public AddCheckCodeAction() {
             return Collections.emptyList();
         }
 
-        Optional<TypeSymbol> foundType = positionDetails.diagnosticProperty(
-                DiagBasedPositionDetails.DIAG_PROP_INCOMPATIBLE_TYPES_FOUND_SYMBOL_INDEX);
+        Optional<Integer> propertyIndex =
+                positionDetails.getPropertyIndex(diagnostic.diagnosticInfo().code(),
+                        DiagnosticPropertyKey.DIAG_PROP_INCOMPATIBLE_TYPES_FOUND);
+        if (propertyIndex.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        Optional<TypeSymbol> foundType = positionDetails.diagnosticProperty(propertyIndex.get());
         if (foundType.isEmpty()) {
             return Collections.emptyList();
         }
```

Comment:
These lines seem to be duplicated, shall we fix that?

Reply:
Done